### ISSUE TITLE
[4.0] Move executable code out of index.php files

### ISSUE THREAD

**Issue by mbabker** [joomla/joomla-cms#14174]
Similar to how we've already restructured the install app, the `administrator/index.php` and `index.php` files should be restructured in a way that allows those files to remain PHP 5.2 friendly while not impeding us using PHP 5.3+ features (like namespaced classes).

**Comment by Spudley** [comment]
Why would we need those files to "remain PHP 5.2 friendly", given that the very first thing those files do is check for PHP 5.3+?

**Comment by zero-24** [comment]
The problem is if they are not  PHP 5.2 friendly they can't be execued by PHP 5.2 so the check is not execued too. 😄

**Comment by mbabker** [comment]
Exactly.  We still want to show a "your platform isn't supported" message to older PHP versions without the files being written in a way that makes them not parseable on older versions.  So either everything in the index.php file has to remain in the global namespace or we give everyone who's on PHP 5.2 (or up to 5.5.8 depending on how the files are written and what language features are used) a PHP fatal because of the invalid syntax for those versions.

Basically, the index.php files should be simplified to this - https://github.com/joomla/joomla-cms/blob/4.0-dev/installation/index.php

**Comment by zero-24** [comment]
@mbabker please review: https://github.com/joomla/joomla-cms/pull/14236
